### ISSUE TITLE
Fine-Tuning Optional cTF-IDF Representations After Model Fit?

### ISSUE THREAD

**Issue by MarkWClements** [MaartenGr/BERTopic#1135]
I am training the BERTopic model on ~1.4 million documents, which takes some time to finish (about 12 hours using "all-MiniLM-L6-v2"). I would like to try variations of fine-tuning the cTF-IDF vectors learned by the model using, e.g.,  `KeyBERTInspired`, `PartOfSpeech`, `MaximalMarginalRelevance`, and `TextGeneration` to get more "coherent" topic representations. However, I don't want to wait 12 hours for the model to run to test each fine-tuning method on my use case. Is there a way to train the topic model without the optional representation and THEN apply various fine-tuning methods to see what the topics look like after each one? This way I can evaluate them without waiting for the model to be fit each time. Pseudo code would look something like this:

```
from bertopic.representation import KeyBERTInspired
from bertopic.representation import MaximalMarginalRelevance
from bertopic import BERTopic

# Train the topic model without optional representations of the cTF-IDF vectors
topic_model = BERTopic(representation_model=None)
topics, probs = topic_model.fit_transform(docs)

# Save the topic model
topic_mode.save('my_bert_model.bert')

# Load the topic model to fine-tune
topic_model = BERTopic.load('my_bert_model.bert')

# Try fine-tuning using KeyBert
representation_model = KeyBERTInspired()
<fine-tune the loaded topic model using KeyBert, save fine-tuned model as topic_model_keybert>

# Try fine-tuning using MaximalMarginalRelevance
representation_model = MaximalMarginalRelevance(diversity=0.3)
<fine-tune the loaded topic model using MMR, save fine-tuned model as topic_model_mmr>
```

Also, is there any guidance on when to choose and/or combine in a chain the respective optional cTF-IDF representations? Specifically between  `KeyBERTInspired`, `PartOfSpeech`, and `MaximalMarginalRelevance`?

Thank you

**Comment by MaartenGr** [comment]
> Is there a way to train the topic model without the optional representation and THEN apply various fine-tuning methods to see what the topics look like after each one?

There definitely is! Big part of BERTopic is its modular nature in such a way that iteration should be fast. More specifically, a major advantage of BERTopic over other techniques is that it separates the clustering with representing the topics. As a result, this allows users to fine-tune the topic representations without the need to re-run the model. 

You can find more about that [here](https://maartengr.github.io/BERTopic/getting_started/topicrepresentation/topicrepresentation.html#update-topic-representation-after-training) but in your case, it would be close to the following:

```python
topic_model.update_topics(docs, representation_model=representation_model)
```

That will update the topic representations with your chosen representation model without re-training the entire model. You can do that as many times as you want to quickly test out which representation models suit your use case best. Also note that you can [chain representation models](https://maartengr.github.io/BERTopic/getting_started/representation/representation.html#chain-models) by, for example, sequentially running `MMR` and `KeyBERTInspired`.

**Comment by MaartenGr** [comment]
Due to inactivity, I'll be closing this issue. Let me know if you want me to re-open the issue!
